util/eventbus/subscribe.go

```diff
@@ -81,16 +94,74 @@ func (q *Queue) pump(stop goroutineShutdownWorker) {
 	}
 }
 
-// A Subscriber delivers one type of event from a [Queue].
+func (s *subscribeState) addSubscriber(t reflect.Type, sub subscriber) {
+	s.outputsMu.Lock()
+	defer s.outputsMu.Unlock()
+	if s.outputs[t] != nil {
+		panic(fmt.Errorf("double subscription for event %s", t))
+	}
+	s.outputs[t] = sub
+	s.client.addSubscriber(t, s)
+}
+
+func (s *subscribeState) deleteSubscriber(t reflect.Type) {
+	s.outputsMu.Lock()
+	defer s.outputsMu.Unlock()
+	delete(s.outputs, t)
+	s.client.deleteSubscriber(t, s)
+}
+
+func (q *subscribeState) subscriber(val any) subscriber {
```

Comment:
maybe `subscriberFor` to avoid the confusion I just had between the constructor of said subscribers and the access of existing ones?

Reply:
Done.